test/jdk/java/awt/Modal/NativeDialogToFrontBackTest.java

```diff
@@ -0,0 +1,117 @@
+ * @run main/manual NativeDialogToFrontBackTest
+ */
+
+public class NativeDialogToFrontBackTest {
+    public static void main(String[] args) throws Exception {
+        final String INSTRUCTIONS = """
+                When the test starts two frames appear: 'Control' and 'Blocked'
+                    1. Click on the 'Show file dialog' button
+                    2. Drag the file dialog so it partially overlaps the 'Blocked' frame
+                    3. 'Blocked' frame must be below the file dialog, if not - press Fail
+                    3. Click on the 'Blocked to front' button
+                    4. 'Blocked' frame must still be below the file dialog, if not - press Fail
+                    5. Close the file dialog
+                    6. Repeat steps 2 to 4 with print and page dialogs using the corresponding button
+                    7. If 'Blocked' frame is always below File/Print/Page dialog, press Pass""";
+
+        PassFailJFrame.builder()
+                .instructions(INSTRUCTIONS)
+                .columns(45)
+                .testUI(NativeDialogToFrontBackTest::init)
+                .build()
+                .awaitAndCheck();
+    }
+
+    public static Frame init() {
+        Frame blocked = new Frame("Blocked");
+        blocked.setBounds(500, 100, 200, 200);
+
+        Frame control = new Frame("Control");
+        control.setModalExclusionType(Dialog.ModalExclusionType.APPLICATION_EXCLUDE);
+        control.setLayout(new FlowLayout());
+
+        Button showFileDialog = new Button("Show file dialog");
+        showFileDialog.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e)
+            {
+                new FileDialog(control, "File dialog").setVisible(true);
+            }
+        });
+        control.add(showFileDialog);
+
+        Button showPrintDialog = new Button("Show print dialog");
+        showPrintDialog.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e)
+            {
+                PrinterJob.getPrinterJob().printDialog();
+            }
+        });
+        control.add(showPrintDialog);
+
+        Button showPageDialog = new Button("Show page dialog");
+        showPageDialog.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e)
+            {
+                PrinterJob.getPrinterJob().pageDialog(new PageFormat());
+            }
+        });
+        control.add(showPageDialog);
+
+        Button blockedToFront = new Button("Blocked to front");
+        blockedToFront.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e)
+            {
+                blocked.toFront();
+            }
+        });
+        control.add(blockedToFront);
+
+        control.setSize(200, 200);
+        blocked.setVisible(true);
+        return control;
```

Comment:
Can u send the Frame list to PassFailJFrame and placement also can be handled their.

Reply:
Updated.